src/map/script-fun.cpp

```diff
@@ -2693,6 +2693,24 @@ void builtin_npctalk(ScriptState *st)
     }
 }
 
+static
+void builtin_walk(ScriptState *st)
+{
+    dumb_ptr<npc_data> nd;
+    short to_x = conv_num(st, &AARG(0));
+    short to_y = conv_num(st, &AARG(1));
+    dumb_ptr<npc_data> nd_ = map_id_is_npc(st->oid);
+    assert (nd_ && nd_->npc_subtype == NpcSubtype::SCRIPT);
+    nd = nd_->is_script();
+
+    nd->to_x = to_x;
+    nd->to_y = to_y;
+    nd->bl_x = to_x;
```

Comment:
my problem is here. `bl_x` and `bl_y` should not be set to the final destination yet, it should be set one tile at a time so it needs to have the same kind of timer as mobs and I got no clue how to do that